### PR TITLE
Grav Hover: Ensure user object is present before using

### PR DESCRIPTION
**Pull request by kraftbj**
Fixes `PHP Notice:  Trying to get property of non-object in ...jetpack/modules/gravatar-hovercards.php on line 238`

Ensure that the $user object is not false before using and return out early if we don't have a hash.

cc: @donnchawp

### CODE REVIEW

File: modules/gravatar-hovercards.php [hunk 1]
```diff
@@ -233,12 +233,19 @@ function grofiles_extra_data() {
...
 function grofiles_hovercards_data_html( $author ) {
 	$data = grofiles_hovercards_data( $author );
+	$hash = '';
 	if ( is_numeric( $author ) ) {
 		$user = get_userdata( $author );
-		$hash = md5( $user->user_email );
+		if ( $user ) {
+			$hash = md5( $user->user_email );
+		}
 	} else {
```
Comment (georgestephanis) [inline modules/gravatar-hovercards.php]: Should this maybe be an `} elseif ( is_email( $author ) ) {` ?

https://codex.wordpress.org/Function_Reference/is_email

**Comment by georgestephanis** [comment]
Previously, #7906 but I think this is a more readable implementation?

**Comment by georgestephanis** [comment]
It also may be worth adding a third conditional for `is_a( 'WP_User' )` in addition to user_id and user_email

**Comment by kraftbj** [comment]
@georgestephanis Made both suggestions, ready for re-review.

**Pull request merged**